registries/adapters/apiv2_adapter.go

```diff
@@ -262,5 +263,43 @@ func (r APIV2Adapter) loadSpec(imageName string) (*bundle.Spec, error) {
 		registryName = fmt.Sprintf("%s:%s", r.config.URL.Hostname(), r.config.URL.Port())
 	}
 
-	return imageToSpec(body, fmt.Sprintf("%s/%s:%s", registryName, imageName, r.config.Tag))
+	schemaVersion, err := getSchemaVersion(body)
+	if err != nil {
+		return nil, err
+	}
+
+	switch schemaVersion {
```

Comment:
Why can't this entire block move into responseToSpec? that way everyone can get this behavior

Reply:
@dymurray actually asked me the same thing...and I started out implementing it that way. However, I didn't see a good way to have the authenticated client available in responseToSpec, and I this functionality is specific only to API V2 compliant registries.